xgcm/test/datasets.py

```diff
@@ -238,3 +238,176 @@ def periodic_2d(request):
+
+
+def datasets_grid_metric(grid_type):
+    """Uniform grid test dataset.
+    Should eventually be extended to nonuniform grid"""
+    xt = np.arange(4)
+    xu = xt + 0.5
+    yt = np.arange(5)
+    yu = yt + 0.5
+    zt = np.arange(6)
+    zw = zt + 0.5
+    t = np.arange(10)
+
+    def data_generator():
+        return np.random.rand(len(xt), len(yt), len(t), len(zt))
+
+    # Need to add a tracer here to get the tracer dimsuffix
+    tr = xr.DataArray(
+        data_generator(), coords=[("xt", xt), ("yt", yt), ("time", t), ("zt", zt)]
+    )
+
+    u_b = xr.DataArray(
+        data_generator(), coords=[("xu", xu), ("yu", yu), ("time", t), ("zt", zt)]
+    )
+
+    v_b = xr.DataArray(
+        data_generator(), coords=[("xu", xu), ("yu", yu), ("time", t), ("zt", zt)]
+    )
+
+    u_c = xr.DataArray(
+        data_generator(), coords=[("xu", xu), ("yt", yt), ("time", t), ("zt", zt)]
+    )
+
+    v_c = xr.DataArray(
+        data_generator(), coords=[("xt", xt), ("yu", yu), ("time", t), ("zt", zt)]
+    )
+
+    wt = xr.DataArray(
+        data_generator(), coords=[("xt", xt), ("yt", yt), ("time", t), ("zw", zw)]
+    )
+
+    # maybe also add some other combo of x,t y,t arrays....
+    timeseries = xr.DataArray(np.random.rand(len(t)), coords=[("time", t)])
+
+    # northeast distance
+    dx = 0.3
+    dy = 2
+    dz = 20
+
+    dx_ne = xr.DataArray(
+        np.ones([len(xt), len(yt)]) * dx - 0.1, coords=[("xu", xu), ("yu", yu)]
+    )
+    dx_n = xr.DataArray(
+        np.ones([len(xt), len(yt)]) * dx - 0.2, coords=[("xt", xt), ("yu", yu)]
+    )
+    dx_e = xr.DataArray(
+        np.ones([len(xt), len(yt)]) * dx - 0.3, coords=[("xu", xu), ("yt", yt)]
+    )
+    dx_t = xr.DataArray(
+        np.ones([len(xt), len(yt)]) * dx - 0.4, coords=[("xt", xt), ("yt", yt)]
+    )
+
+    dy_ne = xr.DataArray(
+        np.ones([len(xt), len(yt)]) * dy + 0.1, coords=[("xu", xu), ("yu", yu)]
+    )
+    dy_n = xr.DataArray(
+        np.ones([len(xt), len(yt)]) * dy + 0.2, coords=[("xt", xt), ("yu", yu)]
+    )
+    dy_e = xr.DataArray(
+        np.ones([len(xt), len(yt)]) * dy + 0.3, coords=[("xu", xu), ("yt", yt)]
+    )
+    dy_t = xr.DataArray(
+        np.ones([len(xt), len(yt)]) * dy + 0.4, coords=[("xt", xt), ("yt", yt)]
+    )
+
+    # dz elements at horizontal tracer points
+    dz_t = xr.DataArray(
+        data_generator() * dz, coords=[("xt", xt), ("yt", yt), ("time", t), ("zt", zt)]
+    )
+    dz_w = xr.DataArray(
+        data_generator() * dz, coords=[("xt", xt), ("yt", yt), ("time", t), ("zw", zw)]
+    )
+    # dz elements at velocity points
+    dz_w_ne = xr.DataArray(
+        data_generator() * dz, coords=[("xu", xu), ("yu", yu), ("time", t), ("zw", zw)]
+    )
+    dz_w_n = xr.DataArray(
+        data_generator() * dz, coords=[("xt", xt), ("yu", yu), ("time", t), ("zw", zw)]
+    )
+    dz_w_e = xr.DataArray(
+        data_generator() * dz, coords=[("xu", xu), ("yt", yt), ("time", t), ("zw", zw)]
+    )
```

Comment:
@jbusecke these are the added `dz` elements. The only difference between your and my version of `datasets.py` is this block and where they are added to metrics below. 

It seems like this should be good to go, and I'll just wait for the other PR's in line to get merged, then I can rebase it to master before merging... but let me know what you think! Thanks for all the rstar references!